java-checks/src/main/java/org/sonar/java/checks/security/EmptyDatabasePasswordCheck.java

```diff
@@ -63,13 +66,18 @@ private static boolean hasEmptyValue(ExpressionTree expression) {
     return literal != null && literal.trim().isEmpty();
   }
 
-  private void checkForUrlConnection(MethodInvocationTree mit) {
+  private static boolean urlContainsEmptyPassword(MethodInvocationTree mit) {
     ExpressionTree urlArgument = mit.arguments().get(URL_ARGUMENT);
     String url = IdentifierUtils.getValue(urlArgument, ConstantUtils::resolveAsStringConstant);
-    if (url != null && (!url.contains("password") || EMPTY_PASSWORD_PATTERN.matcher(url).matches())) {
-      reportIssue(mit, MESSAGE);
-    }
+    return url != null &&
+      (patternContainsEmptyPassword(url, MYSQL_URL_PATTERN) ||
+        patternContainsEmptyPassword(url, ORACLE_URL_PATTERN) ||
+        EMPTY_PASSWORD_PATTERN.matcher(url).matches());
+  }
 
+  private static boolean patternContainsEmptyPassword(String url, Pattern mysqlUrlPattern) {
```

Comment:
Naming:
* Why not name the method `urlContainsEmptyPassword`?
* Why does the parameter name start with `mysql`?

Reply:
- Agreed, going to rename the method name
- Forgot to rename the parameter name proposed by IntelliJ, thanks for pointing that out